pkg/asset/installconfig/vsphere/permissions.go

```diff
@@ -162,14 +167,44 @@ var permissions = map[permissionGroup]PermissionGroupDefinition{
 	},
 }
 
+// pruneToAvailablePermissions different versions of vCenter support different privileges.  the intent of this method
+// is to prune privileges from the check that don't exist.
+func pruneToAvailablePermissions(ctx context.Context, manager AuthManager) error {
+	var authManagerMo mo.AuthorizationManager
+	err := manager.Properties(ctx, manager.Reference(), []string{"privilegeList"}, &authManagerMo)
+	if err != nil {
+		return err
+	}
+	for permissionGroupKey, permissionGroup := range permissions {
+		var prunedPermissions []string
+		for _, permission := range permissionGroup.Permissions {
+			keep := false
+			for _, availablePermission := range authManagerMo.PrivilegeList {
+				if availablePermission.PrivId == permission {
+					keep = true
+					break
+				}
+			}
+			if keep {
+				prunedPermissions = append(prunedPermissions, permission)
+			}
+		}
```

Comment:
What about
```suggestion
    availablePermissions := sets.NewString()
    for _, perm := range authManagerMo.PrivilegeList {
        availablePermissions.Insert(perm.PrivId)
    }
	for permissionGroupKey, permissionGroup := range permissions {
		prunedPermissions := sets.NewString(permissionGroup.Permissions...).Intersection(availablePermissions)
```

Reply:
great suggestion, i'll check that out :)